homeassistant/helpers/event.py

```diff
@@ -1362,6 +1362,8 @@ def async_call_later(
     delay: float | timedelta,
     action: HassJob[[datetime], Coroutine[Any, Any, None] | None]
     | Callable[[datetime], Coroutine[Any, Any, None] | None],
+    *,
+    cancel_on_shutdown: bool = False,
```

Comment:
I think we should require passing in a hassjob and have the flag set on the hassjob at construction time instead of mutating it later.

It might be confusing and create a hard to track down bug where something is canceled when it's not expected to be if a hassjob is passed around and something flips the flag later when it's passed here.